packages/components/src/custom-select-control/index.js

```diff
@@ -147,12 +148,15 @@ export default function CustomSelectControl( {
 					{ label }
 				</StyledLabel>
 			) }
-			<InputBase
+			<InputBaseWithBackCompatMinWidth
+				__nextUnconstrainedWidth={ __nextUnconstrainedWidth }
```

Comment:
I believe there is also a need for `__next36pxDefaultSize` to be passed to `InputBaseWithBackCompatMinWidth` , since the underlying [`InputBase` accepts it as a prop](https://github.com/WordPress/gutenberg/blob/f4dde1cca860fb74bf8b7a2644175bb06bae3102/packages/components/src/input-control/input-base.tsx#L57) and uses it to set the correct prefix/suffix padding?